src/activity_handlers.cpp

```diff
@@ -474,417 +474,131 @@ int butcher_time_to_cut( const player &u, const item &corpse_item, const butcher
-        if( corpse->has_flag( MF_CBM_TECH ) ) {
-            butcher_cbm_item( "bio_power_storage", p->pos(), age, roll_butchery() );
-            butcher_cbm_group( "bionics_tech", p->pos(), age, roll_butchery() );
-        }
-
-        // Substation mini-boss bionics
-        if( corpse->has_flag( MF_CBM_SUBS ) ) {
-            butcher_cbm_item( "bio_power_storage", p->pos(), age, roll_butchery() );
-            butcher_cbm_group( "bionics_subs", p->pos(), age, roll_butchery() );
-            butcher_cbm_group( "bionics_subs", p->pos(), age, roll_butchery() );
-        }
-
-        // Payoff for butchering the zombie bio-op
-        if( corpse->has_flag( MF_CBM_OP ) ) {
-            butcher_cbm_item( "bio_power_storage_mkII", p->pos(), age, roll_butchery() );
-            butcher_cbm_group( "bionics_op", p->pos(), age, roll_butchery() );
-        }
-
-        //Add a chance of CBM power storage recovery.
-        if( corpse->has_flag( MF_CBM_POWER ) ) {
-            butcher_cbm_item( "bio_power_storage", p->pos(), age, roll_butchery() );
-        }
-    }
-
-    // feedback that this type of corpse has implants that can be potentialy removed
-    if( corpse->has_flag( MF_CBM_CIV ) || corpse->has_flag( MF_CBM_SCI ) ||
-        corpse->has_flag( MF_CBM_TECH ) ||
-        corpse->has_flag( MF_CBM_SUBS ) || corpse->has_flag( MF_CBM_OP ) ||
-        corpse->has_flag( MF_CBM_POWER ) ) {
-        if( action == F_DRESS ) {
-            p->add_msg_if_player( m_bad,
-                                  _( "You suspect there might be bionics implanted in this corpse, that careful dissection might reveal." ) );
-        }
-        if( action == BUTCHER || action == BUTCHER_FULL ) {
-            switch( rng( 1, 3 ) ) {
-                case 1:
-                    p->add_msg_if_player( m_bad,
-                                          _( "Your butchering tool encounters something implanted in this corpse, but your rough cuts destroy it." ) );
-                    break;
-                case 2:
-                    p->add_msg_if_player( m_bad,
-                                          _( "You find traces of implants in the body, but you care only for the flesh." ) );
-                    break;
-                case 3:
-                    p->add_msg_if_player( m_bad,
-                                          _( "You found some bionics in the body, but harvesting them would require more surgical approach." ) );
-                    break;
-            }
-        }
-    }
-
-    //now handle the meat, if there is any
-    if( meat != "null" ) {
-        if( pieces <= 0 ) {
-            if( action == BUTCHER || action == BUTCHER_FULL ) {
-                p->add_msg_if_player( m_bad, _( "Your clumsy butchering destroys the flesh!" ) );
-            }
+// The below function exists to allow mods to migrate their content fully to the new harvest system. This function should be removed eventually.
+harvest_list butchery_flags_deprecate( const mtype &mt )
```

Comment:
Elsewhere in the codebase uses the term 'legacy' for this sort of backwards compatibility.  Maybe name this function accordingly.  Something like `harvest_for_legacy_mtype`?

Reply:
is legacy stuff intended to be removed in the future? that was my plan for this function, while giving some leeway for modders to update their monsters

Reply:
legacy stuff doesn't get removed if it would affect savegame compatibility.  We're less concerned about mods.

Reply:
it won't affect vanilla save game compatibility. the new harvest entries are getting updated at the same time as removing the flags from the monsters; the only reason this function is here is to give leeway for modders.